#162948910 Users should be able to reset password via email

boltonotieno opened the pull request:
#### What does this PR do?

Allows the user to reset their password 

#### Description of Task to be completed?

* User should request to reset their password
* A  link should be sent to their email address
* From the link, they should be able to reset their password
#### How should this be manually tested?

* Clone ah-alpha repo $ git clone https://github.com/andela/ah-alpha.git
* git checkout `ft-user-reset-password-162948910`
* Run `pip install -r requirements.txt` to update requirements.txt 
* set up postgres database on the psql command line
* Create a virtual environment and create a .env file and add the database credentials
* export database credentials. please note that there are no spaces before and after =
``` export DB_NAME=" name of your database"
      export DB_PASSWORD="passord to your database user"
      export DB_HOST="your local server eg (127.0.0.1)"
      export DB_USER="your database username eg (postgres)"
```
* Note- if you have no password set for your database leave the password field blank example: 
  `export DB_PASSWORD=""`

* In the project's root directory run `python manage.py runserver`
* Create a user by sending a POST request to the url http://127.0.0.1:8000/api/users/ with the following in the body.
```
{
    "user":{
        "username":"Username",
        "email":"example@email.com",
        "password":"Password"
    }
}
```
* Request for your password reset by sending a POST request to  the link http://127.0.0.1:8000/api/users/password_request/ with the following in the body.
```
{
    "user":{
        "email":"your email"
    }
}
```
- You will receive a link to reset your password in your email address
- copy the link and paste on postman
- send a PATCH request to the copied link 
- In the body tag use the format below to change your password
```
{
    "user":{
        "password":new password"
    }
}
```
#### What are the relevant pivotal tracker stories?
[#162948910 ](https://www.pivotaltracker.com/story/show/162948910 )

tesh254 commented on authors/apps/authentication/backends.py:
```diff
@@ -1,10 +1,36 @@
+"""Configure JWT Here"""
+
+
+class GetAuthentication(TokenAuthentication):
```
Introduce doc strings for every method, function and class

Ruiru11 replied:
On it

tesh254 commented on authors/apps/authentication/models.py:
```diff
@@ -98,16 +101,16 @@ def __str__(self):
+        return self.username
 
     def get_short_name(self):
         """
```
Reduce this space here

Ruiru11 replied:
Working on this

boltonotieno committed:
ft(reset password):A user should be able to reset their password

- First a user requests for passowrd reset
- Then an email is sent to the users email
- From the email sent a user can change their password
- test send request and password reset

[Delivers #162948910]

frankip commented on authors/apps/authentication/models.py:
```diff
@@ -4,117 +4,121 @@
 )
-from django.db import models 
+from django.db import models
+# from rest_framework_jwt.settings import api_settings
```
@BoltC0rt3z remove unused code

boltonotieno replied:
Its done

frankip commented on authors/apps/authentication/tests/test_password_request.py:
```diff
@@ -0,0 +1,116 @@
+
+        self.assertEqual(
+            response.json(),
+            {"message": 'Please provide an email'})
```
@BoltC0rt3z Nice work on testing the error messages but you can import the messages file to assert the response messages in your tests

boltonotieno replied:
Thank you for the feedback. I have worked on it.
